src/index.ts

```diff
@@ -569,3 +585,40 @@ const main = async ({
     })
   );
 })();
+
+type PackageManager = "npm" | "pnpm" | "yarn";
+
+async function exists(path) {
+  try {
+    await access(path);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function getPkgManager(): Promise<PackageManager> {
+  // Use npm_config_user_agent if it's set
+  const userAgent = process.env.npm_config_user_agent;
+
+  if (userAgent) {
+    if (userAgent.startsWith("yarn")) {
+      return "yarn";
+    } else if (userAgent.startsWith("pnpm")) {
+      return "pnpm";
+    } else {
+      return "npm";
+    }
+  }
+
+  const hasYarnLock = await exists("yarn.lock");
+  const hasPnpmLock = await exists("pnpm-lock.yaml");
```

Comment:
For [PNPM workspaces](https://pnpm.io/workspaces) it's possible we'd want to traverse up until we find a `pnpm-lock.yaml` which would be at the workspace root.

Alternatively package manager could be an option ... _or_ we could just add some documentation asking users to set `npm_config_user_agent`.

I'm in different / am happy to tweak this PR in any way